basic_autonomy/include/basic_autonomy/helper_functions.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+/*
+ * Copyright (C) 2021 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include <basic_autonomy/basic_autonomy.h>
+
+namespace basic_autonomy
+{
+namespace waypoint_generation
+{
+    
+    /**
+     * \brief Returns the nearest point to the provided vehicle pose in the provided list
+     * 
+     * \param points The points to evaluate
+     * \param state The current vehicle state
+     * 
+     * \return index of nearest point in points
+     */
+    int get_nearest_point_index(const std::vector<lanelet::BasicPoint2d>& points,
+                                                 const cav_msgs::VehicleState& state);
+    /**
+     * \brief Returns the nearest point to the provided vehicle pose in the provided list
+     * 
+     * \param points The points to evaluate
+     * \param state The current vehicle state
+     * 
+     * \return index of nearest point in points
+     */
+    int get_nearest_point_index(const std::vector<PointSpeedPair>& points,
+                                                 const cav_msgs::VehicleState& state);
+    /**
+     * \brief Returns the nearest point to the provided vehicle pose in the provided list by utilizing the downtrack measured along the route
+     * This function compares the downtrack, provided by routeTrackPos, of each points in the list to get the closest one to the given downtrack.
+     * 
```

Comment:
Should have same NOTE: section as the other two downtrack methods

Reply:
added the note